### DISCUSSION TITLE
How to send events by batch to Google Calendar?

### DISCUSSION THREAD

**Discussion by RyanSacks** [spatie/laravel-google-calendar#236] [Q&A]
Hi,
This is just a question, not an issue, sorry, but am I able to insert multiple events by batch to Google Calendar? Or do I need to do them one by one? And if I need to do them one by one, I'm assuming I can create a job for this and add a queue worker. Would that be an acceptable way to go about it? Thank you for your help.

**Comment by tomcodes** [comment]
Probably late to the party but currently this package does not support sending batch requests.

Creating a job + queue worker (or Horizon) works well for this purpose, just be aware that you might get some rate limiting from Google API depending on the speed of your API calls.

From my experience, updating let's say 365 events in a row (for instance for a daily recurrent event for 1 year) kinda works if you manage to delay a bit your jobs

![image](https://user-images.githubusercontent.com/2213846/184108600-c6043aa0-94bf-42a0-9792-af523765c774.png)

You'd want to stay below 5 requests / second so you should delay a bit your job dispatch.

Here is the documentation about API rate limiting:  
https://developers.google.com/calendar/api/guides/quota

As for implementing batch requests in this package here is the documentation to send batch requests:  
https://developers.google.com/calendar/api/guides/batch
